taxi/views.py

```diff
@@ -26,94 +26,154 @@ def index(request):
         "num_manufacturers": num_manufacturers,
         "num_visits": num_visits + 1,
     }
+    return render(request, "taxi/index.html",
+                  context=context)
+
+
+class BaseListView(LoginRequiredMixin,
+                   generic.ListView):
+    paginate_by = 5
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        query = self.request.GET.get("q")
+        if query:
+            queryset = queryset.filter(self.get_search_filter(query))
+        print(queryset)
```

Comment:
The use of 'print(queryset)' is not recommended in production code as it can lead to performance issues and clutter the console output. Consider removing this line or using logging if needed for debugging.